DataFormats/MuonReco/src/Muon.cc

```diff
@@ -334,6 +334,78 @@ unsigned int Muon::stationGapMaskPull( float sigmaCut ) const
    return totMask;
 }
 
+int Muon::nDigisInStation( int station, int muonSubdetId ) const
+{
+  int nDigis(0);
+  std::map<int, int> me11DigisPerCh;
+
+  if ( muonSubdetId != MuonSubdetId::CSC  &&
+       muonSubdetId != MuonSubdetId::DT )
+    return 0;
+	   
+  for ( auto & match : muMatches_ )
+    {
+      if ( match.detector() != muonSubdetId ||
+	   match.station()  != station )
+	continue;
+	  
+      int nDigisInCh = match.nDigisInRange;
+      
+      if( muonSubdetId == MuonSubdetId::CSC && station == 1 )
+	{
+	  CSCDetId id(match.id.rawId());
+	  
+	  int chamber = id.chamber();
+          int ring    = id.ring();
+	  
+	  if ( ring == 1 || ring == 4 ) // merge ME1/1a and ME1/1b digis
+	    {
+	      if( me11DigisPerCh.find(chamber) == me11DigisPerCh.end() )
+		me11DigisPerCh[chamber] = 0;
+	      
+	      me11DigisPerCh[chamber] += nDigisInCh;
+	      
+	      continue;
+	    }
+	}
+      
+      if( nDigisInCh > nDigis )
+	nDigis = nDigisInCh;
+    }
+
+  for ( const auto & me11DigisInCh : me11DigisPerCh )
+    {  
+      int nMe11DigisInCh = me11DigisInCh.second;
+      if ( nMe11DigisInCh > nDigis )
+	nDigis = nMe11DigisInCh;
+    }
+  
+  return nDigis;
+}
+
+bool Muon::hasShowerInStation( int station, int muonSubdetId, int nDtDigisCut, int nCscDigisCut ) const
+{
+  bool hasShower = muonSubdetId == MuonSubdetId::DT ? 
+    nDigisInStation(station,muonSubdetId) >= nDtDigisCut :
+    nDigisInStation(station,muonSubdetId) >= nCscDigisCut;
+
+  return hasShower;   
+}
+
+int Muon::numberOfShowers( int nDtDigisCut, int nCscDigisCut ) const
+{
+  int nShowers = 0;
+  for ( int iCh = 1; iCh < 5; ++iCh )
```

Comment:
to reduce possible confusion when reading this, perhaps ```iSt``` instead of ```iCh``` because it's supposed to be a shorthand for a station

Reply:
Changed in `station` (more consistent with the member functions above).